src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1478 @@
+#include "../config.h"
+#include "../graphics/colours.h"
+#include "../graphics/image_ids.h"
+#include "../input.h"
 #include "../interop/interop.hpp"
+#include "../localisation/FormatArguments.hpp"
+#include "../objects/building_object.h"
+#include "../objects/interface_skin_object.h"
+#include "../objects/objectmgr.h"
 #include "../openloco.h"
+#include "../town.h"
+#include "../townmgr.h"
 #include "../ui/WindowManager.h"
+#include "../ui/scrollview.h"
+#include "../widget.h"
+#include <intrin0.h>
```

Comment:
What's this for? This doesn't look like a standard header, nor one of ours.

Reply:
I was getting a CI error for `_BitScanReverse()` so I tried this as `intrin0.h` is where it was defined, after googling the error. But it didn't work as CI still failed.

Reply:
Bit scan forward is defined in numeric.cpp. I don't think we have a bit scan reverse

Reply:
We will have to create a new function for reverse as it's not a standard function

Reply:
would it look something like this?
```
    int32_t bitscanreverse(uint32_t source)
    {
#if defined(_MSC_VER) && (_MSC_VER >= 1400) // Visual Studio 2005
        DWORD i;
        uint8_t success = _BitScanReverse(&i, source);
        return success != 0 ? i : -1;
#elif defined(__GNUC__)
        int32_t success = __builtin_ffs(__builtin_bswap32(source));
        return success - 1;
#else
#pragma message "Falling back to iterative bitscan reverse, consider using intrinsics"
        for (int32_t i = 31; i > -1 ; i++)
            if (source & (1u << i))
                return i;

        return -1;
#endif
```

Reply:
That looks like alright, yes. Though I wonder if there's a more optimal GCC intrinsic to use.